core/drand_beacon_control.go

```diff
@@ -870,27 +882,38 @@ func (bp *BeaconProcess) Status(c context.Context, in *drand.StatusRequest) (*dr
 	}
 
 	// remote network connectivity
+	bp.log.Debugw("Starting remote network connectivity check", "for_nodes", in.GetCheckConn())
 	resp := make(map[string]bool)
 	for _, addr := range in.GetCheckConn() {
-		if addr.GetAddress() == bp.priv.Public.Addr {
+		remoteAddress := addr.GetAddress()
+		if remoteAddress == "" {
+			bp.log.Warnw("Skipping empty address", "addr", addr)
 			continue
 		}
+		if remoteAddress == bp.priv.Public.Addr {
+			// skipping ourself
+			continue
+		}
+
 		// TODO check if TLS or not
```

Comment:
can remove this comment I think

Reply:
I guess xD 
TLS will need refactoring over the whole codebase anyway.